src/main/kotlin/eli/inevitable/formatters/CellFormatter.kt

```diff
@@ -0,0 +1,261 @@
+            checkFinalizedStatus()
+            buildable.rawLines.clear()
+
+            return addLines(lines)
+        }
+
+        /**
+         * @param lines The [array of text lines][String] to populate the rows of the cell.
+         * @return This [Builder]
+         */
+        fun addLines(vararg lines: String): Builder {
+            return addLines(lines.toList())
+        }
+
+        /**
+         * @param lines The [list of text lines][String] to populate the rows of the cell.
+         * @return This [Builder]
+         */
+        fun addLines(lines: List<String>): Builder {
+            checkFinalizedStatus()
+            with(buildable) {
+                rawLines.addAll(lines.flatMap { it.sanitize() })
+                width = maxOf(width, rawLines.maxWidth() + 2 * horizontalPaddingCount)
+                height = maxOf(height, rawLines.size + 2 * verticalPaddingCount)
+            }
+
+            return this
+        }
+
+        /**
+         * @param alignment Configures how to align the text within the cell. Default: [Align.LEFT]
+         * @return This [Builder]
+         */
+        fun withAlignment(alignment: Align): Builder {
+            checkFinalizedStatus()
+            buildable.alignment = alignment
+
+            return this
+        }
+
+        /**
+         * @param token The displayable [token][Char] used to fill the spacing between the raw-text and the cell padding. Default: ' '.
+         * @return This [Builder]
+         */
+        fun withFillingSpacer(token: Char): Builder {
+            checkFinalizedStatus()
+            buildable.fillingSpacerToken = token
+
+            return this
+        }
+
+        /**
+         * @param token The displayable [token][Char] used to pad the spacing between the block text and the cell borders. Default: ' '.
+         * @return This [Builder]
+         */
+        fun withPaddingSpacer(token: Char): Builder {
+            checkFinalizedStatus()
+            buildable.paddingSpacerToken = token
+
+            return this
+        }
+
+        /**
+         * @param verticalPadding The minimum number of empty rows to space between horizontal cell borders.
+         * @param horizontalPadding The minimum number of empty columns to space between vertical cell borders.
+         * @return This [Builder]
+         */
+        fun withPadding(verticalPadding: Int, horizontalPadding: Int): Builder {
+            checkFinalizedStatus()
+            with(buildable) {
+                verticalPaddingCount = verticalPadding
+                horizontalPaddingCount = horizontalPadding
+            }
+
+            return this
+        }
+
+        /**
+         * @param height The positive total number of lines for the cell. If not set, will size to best fit the contents of the cell. Cannot be set to a value smaller than the contents of the cell including padding.
+         * @return This [Builder]
+         */
+        fun withTotalHeight(height: Int): Builder {
```

Comment:
maybe rename to maxHeight/maxWidth

Reply:
Cannot vs Should not,

I don't see something that would throw an exception to let me know I messed up as part of 'cannot'

Reply:
I like the name adjustment. Will fix.

It won't be set to anything smaller. It won't fail, but it won't let you (takes the larger of the options).